runtime/pallene_core.h

```diff
@@ -226,4 +226,41 @@ int pallene_is_truthy(const TValue *v)
     return !(ttisnil(v) || (ttisboolean(v) && bvalue(v) == 0));
 }
 
+static inline
+Table *pallene_new_table(lua_State *L, lua_Integer n)
+{
+    Table *t = luaH_new(L);
+    if (n > 0) {
+        luaH_resize(L, t, 0, n);
+    }
+    return t;
+}
+
+static const TValue PALLENE_ABSENTKEY = {ABSTKEYCONSTANT};
+
+static inline
+TValue *pallene_getstr(Table *t, TString *key, int *pos)
+{
+    if (PALLENE_LIKELY(*pos < sizenode(t))) {
```

Comment:
This test should also check that pos is non-negative. Use an unsigned comparison, like we do for arrays.

Reply:
Why would pos be negative?

Reply:
As I suggested in other comments, we could use a negative pos to signify that the cache is invalid. But even if we didn't do that, using an unsigned comparison would be an extra layer of defensive programming, at no cost.